### DISCUSSION TITLE
How to Unit test for Linux and windows?

### DISCUSSION THREAD

**Discussion by jeromelaban** [unoplatform/uno#10956] [Q&A]
[_Question from discord_](https://discord.com/channels/372137812037730304/550416151172087808/1062607822744993792)

**Comment by jeromelaban** [comment]
[Uno.UITest](https://github.com/unoplatform/uno.uitest) does not support Linux or Windows at this point, but we're currently working on what we're calling a ["runtime tests" engine](https://github.com/unoplatform/uno.ui.runtimetests.engine). It's unit testing inside the running application, with the ability to manipulate the UI.

It's still in development (and not super pretty) but does the job for our own tooling. Give it a try and let us know!

**Reply by michael-hawker** [reply]
Since the UWP/WinAppSDK MS Test heads are effectively apps, couldn't a similar template be setup to use your new runtime engine with the app shell and then leverage an existing UWP/WinAppSDK test suite?

Basically, thinking we could do this for labs (as we already handle the XAML interop for running on UI Thread and dealing with Content for the app already), and we're working on using the WinRT Input Injection API for more E2E testing.

**Reply by jeromelaban** [reply]
@michael-hawker that could definitely be useful indeed! Input injection is something we're making strides to include in uno targets so that in-app testing becomes more powerful.
